libsoftwarecontainer/unit-test/devicenode_unittest.cpp

```diff
@@ -0,0 +1,249 @@
+
+    ASSERT_TRUE(dev.parse(configJSON));
+    ASSERT_EQ(dev.getMode(), 622);
+    ASSERT_FALSE(dev.getIsconfigured());
+    ASSERT_TRUE(dev.activate(container));
+    ASSERT_TRUE(dev.getIsconfigured());
+}
+
+
+
+struct testSetup
+{
+    Device firstConfiguration;
+    Device secondConfiguration;
+    bool  shouldReconfigure;
+};
+
+class DynamicDeviceConfigurationTests : public  ::testing::TestWithParam<testSetup>
+{
+public:
+    DeviceNodeLogic dnl;
+    std::shared_ptr<ContainerAbstractInterface> container;
+
+    void SetUp() override
+    {
+        container = std::shared_ptr<ContainerAbstractInterface> (new MockContainer());
+    }
+};
+
+/*
+ * This data is fed to the DynamicDeviceConfigurationTests tests
+ *
+ * The first set examines the case when new mode is configured for same device
+ * The second set examines when a new device is added to configuration
+ * The third set examines when exactly same device with same mode is added
+ */
+INSTANTIATE_TEST_CASE_P(DeviceModeConfigurations, DynamicDeviceConfigurationTests, ::testing::Values(
+        testSetup{
+            {"/dev/input/mouse0", 622},
+            {"/dev/input/mouse0", 755},
+            false
+        },
+        testSetup{
+            {"/dev/tty0", 622},
+            {"/dev/tty1", 652},
+            true
+        },
+        testSetup{
+            {"/dev/tty1", 622},
+            {"/dev/tty1", 622},
+            true
+        }
+));
+
+TEST_P(DynamicDeviceConfigurationTests, DoubleActivation) {
+    testSetup testparams = GetParam();
+    //add first device to the device list of DeviceNodeLogic
+    ASSERT_TRUE(dnl.updateDeviceList(testparams.firstConfiguration));
+
+    //configure device
+    auto devlist = dnl.getDevList();
+    for (auto &dev : devlist) {
+        dev->activate(container);
+    }
+
+    //add second device to the device list of DeviceNodeLogic
+    ASSERT_TRUE(dnl.updateDeviceList(testparams.secondConfiguration));
+    auto item = dnl.findDeviceByName(testparams.firstConfiguration.getName());
+    ASSERT_NE(nullptr, item);
+    ASSERT_EQ(testparams.shouldReconfigure, item->getIsconfigured());
+}
+
+
+struct testModes
+{
+    int itemMode;
+    int newMode;
+    int expectedMode;
+};
+
+class DeviceModeTests : public  ::testing::TestWithParam<testModes>
+{
+public:
+    Device dev;
+};
+
+/*
+ * This data is fed to the DeviceMode tests
+ */
+INSTANTIATE_TEST_CASE_P(DeviceModeParameters, DeviceModeTests, ::testing::Values(
+        testModes{644, 777 ,777},
+        testModes{444, 266 ,466},
+        testModes{444, 67  ,467},
+        testModes{644, 422 ,644},
+        testModes{20,  644 ,644},
+        testModes{666, 666 ,666}
+));
+
+
+/*
+ * Verify whitelisting in the case when mode is set to more permissive
+ */
+TEST_P(DeviceModeTests, UpdateDeviceMode) {
+    auto testparams = GetParam();
+    dev.setMode(testparams.itemMode);
+    dev.calculateDeviceMode(testparams.newMode);
+    ASSERT_EQ(testparams.expectedMode, dev.getMode());
+}
```

Comment:
These tests perform the same testing as the component test that's added in this PR does, right? Then I would vote for removing the component test. We should add a service test, but that should not be part of this PR imo.